packages/next/shared/lib/router/router.ts

```diff
@@ -922,6 +922,11 @@ export default class Router implements BaseRouter {
       window.location.href = url
       return false
     }
+    // Decode hash to make non-latin anchor works.
+    // A hash mark # is the optional last part of a URL
+    const hashRegex = /#.+$/
+    as = as.replace(hashRegex, (m) => decodeURIComponent(m))
```

Comment:
Can we move the hash decoding directly to `scrollToHash` and make sure to only decode the `hash` value being used for `getElementById`? I don't think we want to modify the `asPath` and store it in the decoded format.

Reply:
Good point.